components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerAPIResourceManagementService.java

```diff
@@ -0,0 +1,468 @@
+
+    }
+
+    public static ServerAPIResourceManagementService getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Add API resource.
+     *
+     * @param apIResourceCreationModel API resource creation model.
+     * @return Response.
+     */
+    public APIResourceResponse addAPIResourceWithResourceId(APIResourceCreationModel apIResourceCreationModel) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Adding API resource with resource id: " + apIResourceCreationModel.getIdentifier());
+        }
+        try {
+            APIResource apiResource = createAPIResource(apIResourceCreationModel);
+            APIResource createdAPIResource = APIResourceManagementServiceHolder.getApiResourceManager()
+                    .addAPIResource(apiResource, CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+            if (createdAPIResource == null) {
+                LOG.error(ErrorMessage.ERROR_CODE_ADD_API_RESOURCE.getDescription());
+                throw APIResourceMgtEndpointUtil.handleException(Response.Status.INTERNAL_SERVER_ERROR,
+                        ErrorMessage.ERROR_CODE_ADD_API_RESOURCE);
+            }
+            return buildAPIResourceResponse(createdAPIResource);
+        } catch (APIResourceMgtException e) {
+            throw APIResourceMgtEndpointUtil.handleAPIResourceMgtException(e);
+        }
+    }
+
+    /**
+     * Get API Resources.
+     *
+     * @param before - before parameter for cursor based pagination.
+     * @param after  - after parameter for cursor based pagination.
+     * @param filter - filter parameter.
+     * @return Response with API Resources list.
+     */
+    public APIResourceListResponse getAPIResources(String before, String after, String filter, Integer limit) {
+
+        APIResourceListResponse apiResourceListResponse = new APIResourceListResponse();
+
+        try {
+            // Set default values if the parameters are not set.
+            limit = validatedLimit(limit);
+
+            // Validate before and after parameters.
+            if (StringUtils.isNotBlank(before) && StringUtils.isNotBlank(after)) {
+                throw APIResourceMgtEndpointUtil.handleException(Response.Status.BAD_REQUEST,
+                        APIResourceMgtEndpointConstants.ErrorMessage.ERROR_CODE_BOTH_BEFORE_AFTER_PROVIDED);
+            }
+
+            // Set the pagination sort order.
+            String paginationSortOrder = StringUtils.isNotBlank(before) ? DESC_SORT_ORDER : ASC_SORT_ORDER;
+
+            APIResourceSearchResult apiResourceSearchResult = APIResourceManagementServiceHolder.getApiResourceManager()
+                    .getAPIResources(before, after, limit + 1, filter, paginationSortOrder,
+                            CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+            List<APIResource> apiResources = apiResourceSearchResult.getAPIResources();
+
+            if (limit != 0 && CollectionUtils.isNotEmpty(apiResources)) {
```

Comment:
I think `limit != 0` can be removed due to ` validatedLimit(limit);` method invocation